be/src/storage/sstable/persistent_index_sstable.cpp

```diff
@@ -17,65 +17,68 @@
 
 #include "fs/fs.h"
-#include "storage/lake/persistent_index_memtable.h"
-#include "storage/lake/tablet_manager.h"
 #include "storage/sstable/table_builder.h"
 #include "util/trace.h"
 
 namespace starrocks::sstable {
 
-Status PersistentIndexSstable::init(RandomAccessFile* rf, const int64_t filesz, Cache* cache) {
+Status PersistentIndexSstable::init(std::unique_ptr<RandomAccessFile> rf, const PersistentIndexSstablePB& sstable_pb,
+                                    Cache* cache) {
     Options options;
     _filter_policy.reset(const_cast<FilterPolicy*>(NewBloomFilterPolicy(10)));
     options.filter_policy = _filter_policy.get();
     options.block_cache = cache;
     Table* table;
-    RETURN_IF_ERROR(Table::Open(options, rf, filesz, &table));
+    RETURN_IF_ERROR(Table::Open(options, rf.get(), sstable_pb.filesize(), &table));
     _sst.reset(table);
+    _rf = std::move(rf);
+    _sstable_pb.CopyFrom(sstable_pb);
     return Status::OK();
 }
 
-Status PersistentIndexSstable::build_sstable(const phmap::btree_map<std::string, IndexValueWithVer, std::less<>>& map,
-                                             WritableFile* wf, uint64_t* filesz) {
+Status PersistentIndexSstable::build_sstable(
+        const phmap::btree_map<std::string, std::list<IndexValueWithVer>, std::less<>>& map, WritableFile* wf,
+        uint64_t* filesz) {
     std::unique_ptr<FilterPolicy> filter_policy;
     filter_policy.reset(const_cast<FilterPolicy*>(NewBloomFilterPolicy(10)));
     Options options;
     options.filter_policy = filter_policy.get();
     TableBuilder builder(options, wf);
-    for (const auto& [k, m] : map) {
+    for (const auto& [k, v] : map) {
         IndexValueWithVerPB index_value_pb;
-        index_value_pb.add_versions(m.first);
-        index_value_pb.add_values(m.second);
+        for (const auto& index_value_with_ver : v) {
+            index_value_pb.add_versions(index_value_with_ver.first);
+            index_value_pb.add_values(index_value_with_ver.second.get_value());
+        }
         builder.Add(Slice(k), Slice(index_value_pb.SerializeAsString()));
     }
     RETURN_IF_ERROR(builder.Finish());
     *filesz = builder.FileSize();
     return Status::OK();
 }
 
-Status PersistentIndexSstable::multi_get(size_t n, const Slice* keys, const KeyIndexesInfo& key_indexes_info,
-                                         int64_t version, IndexValue* values, KeyIndexesInfo* found_keys_info) {
+Status PersistentIndexSstable::multi_get(size_t n, const Slice* keys, const std::set<KeyIndex>& key_indexes,
+                                         int64_t version, IndexValue* values, std::set<KeyIndex>* found_key_indexes) {
     std::vector<std::string> index_value_infos(n);
     ReadOptions options;
     auto start_ts = butil::gettimeofday_us();
-    RETURN_IF_ERROR(_sst->MultiGet(options, n, keys, key_indexes_info, &index_value_infos));
+    RETURN_IF_ERROR(_sst->MultiGet(options, keys, key_indexes, &index_value_infos));
```

Comment:
didn't get through the detailed implementation of the MultiGet implementation. However, there are two potential issues here:
1. `std::vector<std::string> index_value_infos(n);` initializes `n` std::string objects all at once, this may be a great waste in turns of key missing and in large scale multi-get.
2. `index_values_infos[key_index].empty()` check, it may not necessarily means the key is not found, but could be that the key is found but the value is just an empty string. This is a design decision whether empty string is a valid value or not. If yes, the code here doesn't handle the case correctly.
3. the MultiGet uses the `value` of the key_indexes to put founded value into index_value_infos.  It can be that there is only one key index in key_indexes, lets say `key_indexes = [12]`, the index_value_infos will be created with one element, however, the multiget will try to access `index_value_infos[12]` if found. Is there any assumption to this key_indexes?